packages/fuel-gauge/test/fixtures/forc-projects/bytecode-sway-lib/src/main.sw

```diff
@@ -1,33 +1,25 @@
 contract;
 
-// TODO: Uncomment bytecode-related stuff
-// This requires sway-libs to be compatible with latest forc (0.52+)
-// use bytecode::*;
-
-
 abi MyContract {
     fn compute_bytecode_root(bytecode_input: Vec<u8>);
 
     fn verify_contract_bytecode(contract_id: ContractId, bytecode: Vec<u8>) -> bool;
 
     // fn compute_predicate_address(bytecode: Vec<u8>) -> Address;
-    fn compute_predicate_address(bytecode: Vec<u8>) -> bool;
 }
 
 impl MyContract for Contract {
     fn compute_bytecode_root(bytecode_input: Vec<u8>) {
-        // let mut bytecode = bytecode_input;
-        // let root = compute_bytecode_root(bytecode);
-        // log(root);
+        // simply logs the hexidecimal b256 string of the bytecode input for testing purposes
+        let ret: b256 = 0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20;
+        log(ret);
```

Comment:
Would the documentation lose its meaning if we changed the snippet?


![doc-snippet](https://github.com/FuelLabs/fuels-ts/assets/26660/7b06d465-b696-4749-8cfe-6bce917d852f)

Reply:
That's a good point, but how do you intend to have this mocked if we still rely on the `bytecode` lib from sway libs?

Reply:
Yeah, good question. I didn't think this through enough.

One suggestion @Dhaiwat10 gave was to use a hardcoded Sway snippet, but that would be a temporary solution.

We should reevaluate whether this page is really needed or rethink/rewrite it differently.

Reply:
I don't think this page is particularly relevant to the TS SDK - if we begin to delve into documenting how to work with Sway libraries it would be determine where we draw the line. I think this is under the purview of the Sway library.

Reply:
Well, @Dhaiwat10 can confirm this, but I'd assume it was added because someone had difficulties achieving the described approach. In this case, it is beneficial and valuable to have it there.

However, we need to reframe the message.

 1. New title
 2. Condensed paragraphs
 3. Improved TS snippet readability
 4. Fixed TS imports
 5. Removal of Sway snippet
 6. Pointing to Sway docs instead of source code

In this scenario, your new Sway snippet can stay as is as its only purpose will be to ensure that the Typescript snippet is valid and working. I think that seals the deal.

---

# Converting Bytecode to `Vec<u8>`

Some Sway functions require you to pass in bytecode to the function. The type of the bytecode parameter is usually `Vec<u8>`. For example, take the `compute_bytecode_root` function from the bytecode Sway library ([docs](https://fuellabs.github.io/sway-libs/book/bytecode/index.html?highlight=bytecode#using-the-bytecode-library-in-sway)).

To pass bytecode to this function, you can make use of the `arrayify` function to convert the bytecode file contents into a `UInt8Array`, the Typescript compatible type for Sway's `Vec<u8>` type, and pass it the function like so:

```ts
import { readFileSync } from 'fs';
import { arrayify } from 'fuels';
 
const bytecode = readFileSync(bytecodePath);
const bytecodeAsVecU8 = arrayify(bytecode);
 
const { value: bytecodeRoot } = await bytecodeContract.functions
  .compute_bytecode_root(bytecodeAsVecU8)
  .call();

console.log(`Bytecode root is: ${bytecodeRoot}`);
```